apis/interfaces.go

```diff
@@ -47,3 +47,8 @@ type Listable interface {
 
 	GetListType() runtime.Object
 }
+
+// Annotatable indicates that a particular type applies various annotations.
+type Annotatable interface {
+	AnnotateUserInfo(previous Annotatable, userName string)
```

Comment:
I'd pass in the [UserInfo](https://godoc.org/k8s.io/api/authentication/v1#UserInfo)

Reply:
Hm, from one point of view sure, from another, do we have ideas how to utilize other fields?

Reply:
What if I just want the UID for PII reasons :)